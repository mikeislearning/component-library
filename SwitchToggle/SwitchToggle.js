import React from 'react';
import { Platform, Switch, Text, View } from 'react-native';
import { PropTypes } from 'prop-types';

import { colors } from 'app/theme';

import styles from './SwitchToggle.styles';

export default class SwitchToggle extends React.Component {
  static propTypes = {
    label: PropTypes.string,
    onValueChange: PropTypes.func.isRequired,
    value: PropTypes.bool.isRequired,
  };

  render() {
    const { label, onValueChange, value } = this.props;
    const labelStyles = [styles.label];
    if (value) labelStyles.push(styles.labelActive);

    return (
      <View style={styles.container}>
        <Switch
          onValueChange={onValueChange}
          thumbColor={Platform.OS === 'android' ? colors.white : null}
          trackColor={{ true: colors.green.dark }}
          value={value}
        />
        {label && <Text style={labelStyles}>{label}</Text>}
      </View>
    );
  }
}
