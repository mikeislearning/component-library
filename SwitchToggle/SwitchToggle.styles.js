import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingTop: 20,
  },
  label: {
    color: colors.gray.medium,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 15,
    paddingLeft: 20,
  },
  labelActive: {
    color: colors.black,
  },
});

export default styles;
