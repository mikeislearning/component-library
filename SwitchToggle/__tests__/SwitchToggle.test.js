import React from 'react';
import { render } from 'react-native-testing-library';

import SwitchToggle from '../SwitchToggle';

describe('<SwitchToggle />', () => {
  it('renders correctly', () => {
    const switchToggle = render(<SwitchToggle onValueChange={() => {}} value={true} />);
    expect(switchToggle).toMatchSnapshot();
  });

  it('renders with a label (when passed)', () => {
    const switchToggle = render(
      <SwitchToggle label={'Your profile is visible'} onValueChange={() => {}} value={true} />
    );
    expect(switchToggle).toMatchSnapshot();
  });
});
