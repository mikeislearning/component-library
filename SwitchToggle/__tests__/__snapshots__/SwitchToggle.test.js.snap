// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchToggle /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingTop": 20,
    }
  }
>
  <RCTSwitch
    accessibilityRole="button"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#00BB83"
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    thumbTintColor={null}
    value={true}
  />
</View>
`;

exports[`<SwitchToggle /> renders with a label (when passed) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingTop": 20,
    }
  }
>
  <RCTSwitch
    accessibilityRole="button"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#00BB83"
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    thumbTintColor={null}
    value={true}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#726E67",
          "fontFamily": "MessinaSans-Regular",
          "fontSize": 15,
          "paddingLeft": 20,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
  >
    Your profile is visible
  </Text>
</View>
`;
